src/head.sh

```diff
@@ -1,6 +1,41 @@
 #!/usr/bin/env bash
-declare -a "options=($BOOT_JVM_OPTIONS)"
+
+set -euo pipefail
+
+process_properties() {
+    if [[ -f "$1" ]]; then
```

Comment:
Isn't it complaining because of the `set -u`? Minor, maybe we can name `$1`?

Reply:
"$1" is the first argument passed to the function `process_properties`, i.e. the name of the properties file. So it's guaranteed to be set at all times. Or am I misunderstanding you?